### PR TITLE
Replace annotation by attribute

### PR DESCRIPTION
**Pull request by alamirault**
Replace annotation by attribute :tada:

### CODE REVIEW

File: src/Rule/VersionaddedDirectiveShouldHaveVersion.php [hunk 1]
```diff
@@ -24,13 +24,10 @@
...
- * @Description("Ensure a versionadded directive has a version which follows SemVer.")
- *
- * @ValidExample(".. versionadded:: 3.4")
- *
- * @InvalidExample({".. versionadded::", ".. versionadded:: foo-bar"})
- */
+#[Description('Ensure a versionadded directive has a version which follows SemVer.')]
+#[ValidExample('.. versionadded:: 3.4')]
+#[InvalidExample('.. versionadded::')]
+#[InvalidExample('.. versionadded:: foo-bar')]
```
Comment (alamirault) [inline src/Rule/VersionaddedDirectiveShouldHaveVersion.php:30]: We don't have a lot of ValidExample/InvlidExample, so i prefer repeatable attribute type hinted `string` instead `string|string[]`

(It also simplify a little bit md doc generation)

File: src/Rule/NoBracketsInMethodDirective.php [hunk 2]
```diff
@@ -13,22 +13,18 @@
...
 
-/**
- * @Description("Ensure a :method: directive has special format.")
- *
- * @InvalidExample(":method:`Symfony\\Component\\OptionsResolver\\Options::offsetGet()`")
- *
- * @ValidExample(":method:`Symfony\\Component\\OptionsResolver\\Options::offsetGet`")
- */
+#[Description('Ensure a :method: directive has special format.')]
+#[InvalidExample(':method:`Symfony\\\\Component\\\\OptionsResolver\\\\Options::offsetGet()`')]
```
Comment (alamirault) [inline src/Rule/NoBracketsInMethodDirective.php:26]: Do you know how we can avoid `\\\\` ?

Else rendering is not the same

File: src/Attribute/Rule/InvalidExample.php [hunk 3]
```diff
@@ -11,15 +11,13 @@
...
- */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
 class InvalidExample
 {
-    /**
-     * @var string|string[]
-     */
-    public $value;
+    public function __construct(
+        public string $value,
```
Comment (OskarStark) [inline src/Attribute/Rule/InvalidExample.php:20]:
```suggestion
        public readonly string $value,
```

File: src/Attribute/Rule/ValidExample.php [hunk 4]
```diff
@@ -11,15 +11,13 @@
...
- */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
 class ValidExample
 {
-    /**
-     * @var string|string[]
-     */
-    public $value;
+    public function __construct(
+        public string $value,
```
Comment (OskarStark) [inline src/Attribute/Rule/ValidExample.php:20]:
```suggestion
        public readonly string $value,
```

File: src/Rule/VersionaddedDirectiveShouldHaveVersion.php (hunk 1 above)
Comment (OskarStark) [reply src/Rule/VersionaddedDirectiveShouldHaveVersion.php:30]: I like the repeatable attribute 👍

File: src/Rule/NoBracketsInMethodDirective.php (hunk 2 above)
Comment (OskarStark) [reply src/Rule/NoBracketsInMethodDirective.php:26]: Sorry, I don't understand the problem 🤔

**Pull request merged**